GillianCore/engine/GeneralSemantics/General/GInterpreter.ml

```diff
@@ -587,7 +581,11 @@ struct
                | Some state' -> [ state' ]
                | _ -> [])
              fos)
-    | SpecVar xs -> [ State.add_spec_vars state (Var.Set.of_list xs) ]
+    | FreshSVar x ->
+        let new_svar = Generators.fresh_svar () in
+        let state' = State.add_spec_vars state (SS.singleton x) in
```

Comment:
This seems to be a program variable `x`? If so, it can't go into the `spec_vars`, `v` should go into the spec vars?

Reply:
Oh god you're right!